test.cpp

```diff
@@ -0,0 +1,43 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "catch.hpp"
+#include <stdio.h>
+#include <stdlib.h>
+#include "TDD.h"
+
+TEST_CASE("Test case 1") 
+{
+  int array[2] = {4,5};
+  int size = sizeof(array);
+  REQUIRE(check_readings(array,size) == 1);
+}
+
+
+TEST_CASE("Test case 2") 
+{
+  int array[3] = {4,5,6};
+  int size = sizeof(array);
+  REQUIRE(check_readings(array,size) == 2);
```

Comment:
check_readings method name is not intuitive. It is difficult to say what processing it does.
looking at the implementation, it seems that number of consecutive pairs are being counted.

The requirement is to identify the the number of ranges (set of consecutive numbers) and the number of values within each set.
Please check...